src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -250,6 +253,20 @@ public string ReasonPhrase
 
         internal IISHttpServer Server => _server;
 
+        private bool CheckRequestCanHaveBody()
+        {
+            // Http/1.x requests with bodies require either a Content-Length or Transfer-Encoding header.
+            // Note Http.Sys adds the Transfer-Encoding: chunked header to HTTP/2 requests with bodies for back compat.
+            // Transfer-Encoding takes priority over Content-Length.
+            string transferEncoding = RequestHeaders[HttpKnownHeaderNames.TransferEncoding];
+            if (string.Equals("chunked", transferEncoding?.Trim(), StringComparison.OrdinalIgnoreCase))
```

Comment:
Does Http.sys not trim header values?

Reply:
It probably does, this code is just defensive.